qa/src/test/java/org/opensearch/searchrelevance/bwc/IndexMappingTestHelper.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.searchrelevance.bwc;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.Request;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.Response;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.WarningsHandler;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.xcontent.DeprecationHandler;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+
+/**
+ * Utility class for index mapping BWC tests.
+ * Provides common helper methods for creating, validating, and updating index mappings
+ * during backward compatibility tests.
+ */
+public final class IndexMappingTestHelper {
+
+    public static final String JUDGMENT_CACHE_INDEX = ".plugins-search-relevance-judgment-cache";
+
+    // Old schema (version 0) - without modelId and encodedPromptTemplate fields
+    public static final String OLD_MAPPING = "{"
+        + "\"_meta\": { \"schema_version\": 0 },"
+        + "\"properties\": {"
+        + "  \"id\": { \"type\": \"keyword\" },"
+        + "  \"timestamp\": { \"type\": \"date\", \"format\": \"strict_date_time\" },"
+        + "  \"querySet\": { \"type\": \"keyword\" },"
+        + "  \"documentId\": { \"type\": \"keyword\" },"
+        + "  \"contextFieldsStr\": { \"type\": \"keyword\" },"
+        + "  \"rating\": { \"type\": \"keyword\" }"
+        + "}"
+        + "}";
+
+    // New schema (version 1) - with modelId and encodedPromptTemplate fields
+    public static final String NEW_MAPPING = "{"
+        + "\"_meta\": { \"schema_version\": 1 },"
+        + "\"properties\": {"
+        + "  \"id\": { \"type\": \"keyword\" },"
+        + "  \"timestamp\": { \"type\": \"date\", \"format\": \"strict_date_time\" },"
+        + "  \"querySet\": { \"type\": \"keyword\" },"
+        + "  \"documentId\": { \"type\": \"keyword\" },"
+        + "  \"contextFieldsStr\": { \"type\": \"keyword\" },"
+        + "  \"rating\": { \"type\": \"keyword\" },"
+        + "  \"modelId\": { \"type\": \"keyword\" },"
+        + "  \"encodedPromptTemplate\": { \"type\": \"keyword\" }"
+        + "}"
+        + "}";
+
+    public static final String TEST_DOCUMENT = "{"
+        + "\"id\": \"test-cache-entry\","
+        + "\"timestamp\": \"2024-01-01T00:00:00.000Z\","
+        + "\"querySet\": \"test-query-set\","
+        + "\"documentId\": \"doc-1\","
+        + "\"contextFieldsStr\": \"field1,field2\","
+        + "\"rating\": \"0.85\""
+        + "}";
+
```

Comment:
We should add all these mappings in a file and read it from there. In future if we new fields are introduced in the mapping then we can  directly update in file without changing at code level.